### PR TITLE
Ensure proto packages are stripped from release docs

### PR DESCRIPTION
**Pull request by somewhatabstract**
## Test Instructions
1. `yarn start`
1. `yarn start:prod`
1. Compare

Packages marked `private: true` in their `package.json` should be omitted from the prod variant.

### CODE REVIEW

File: utils/gen-styleguide-prod-config.js [hunk 1]
```diff
@@ -72,36 +72,6 @@ function writeStyleguideConfig(filepath, config) {
...
-            sections.splice(sectionIndex, 1);
-        } else {
-            // And now, we move into the chilren of the section and repeat.
-            sections[sectionIndex].sections = removePrivateSections(
-                sections[sectionIndex].sections,
-            );
-        }
-    }
-    return sections;
-}
```
Comment (somewhatabstract) [inline utils/gen-styleguide-prod-config.js]: This is moved, not changed

File: utils/gen-styleguide-prod-config.js [hunk 2]
```diff
@@ -180,32 +150,55 @@ function maybeGetPackageInfoForSection(section) {
...
+            sections.splice(sectionIndex, 1);
+        } else {
+            // And now, we move into the chilren of the section and repeat.
+            sections[sectionIndex].sections = removePrivateSections(
+                sections[sectionIndex].sections,
+            );
+        }
+    }
+    return sections;
+}
```
Comment (somewhatabstract) [inline utils/gen-styleguide-prod-config.js]: Moved from before the other functions to after. It ended up not being necessary to do so, but I forgot to move it back.

File: utils/gen-styleguide-prod-config.js [hunk 3]
```diff
@@ -180,32 +150,55 @@ function maybeGetPackageInfoForSection(section) {
...
     const info = maybeGetPackageInfoForSection(section);
     if (info) {
         // eslint-disable-next-line no-console
         console.log(
-            `Adding package information for ${
-                section.name
-            }: ${JSON.stringify(info)}`,
+            `Adding package information for ${section.name}: ${JSON.stringify(
+                info,
+            )}`,
```
Comment (somewhatabstract) [inline utils/gen-styleguide-prod-config.js]: Prettier change...wasn't me.

File: utils/gen-styleguide-prod-config.js [hunk 4]
```diff
@@ -215,13 +208,27 @@ for (const section of styleguideConfig.sections) {
             `Design Specification: **${info.design}**\n`,
             currentDescription,
         ];
+        // If this is a prototype, then we don't want this in the prod
+        // documentation.
+        if (info.prototype) {
+            section.private = true;
+        }
```
Comment (somewhatabstract) [inline utils/gen-styleguide-prod-config.js]: This is the important bit.

**Comment by jeresig** [comment]
Deploy preview for *wonder-blocks* ready!

Built with commit f4405a88be027a55117fe72112e3c030d04260fc

https://deploy-preview-204--wonder-blocks.netlify.com

File: utils/gen-styleguide-prod-config.js [hunk 5]
```diff
@@ -180,32 +150,55 @@ function maybeGetPackageInfoForSection(section) {
...
  */
-// See https://www.netlify.com/docs/continuous-deployment/#build-environment-variables
-if (process.env.PULL_REQUEST !== "true") {
-    // If we're not a pull request, then let's iterate over the configuration
-    // and remove private sections.
-    styleguideConfig.sections = removePrivateSections(
-        styleguideConfig.sections,
-    );
-}
-
```
Comment (somewhatabstract) [inline utils/gen-styleguide-prod-config.js]: Moved to occur after processing.

File: utils/gen-styleguide-prod-config.js [hunk 6]
```diff
@@ -215,13 +208,27 @@ for (const section of styleguideConfig.sections) {
...
 }
 
+// See https://www.netlify.com/docs/continuous-deployment/#build-environment-variables
+if (process.env.PULL_REQUEST !== "true") {
+    // If we're not a pull request, then let's iterate over the configuration
+    // and remove private sections.
+    styleguideConfig.sections = removePrivateSections(
+        styleguideConfig.sections,
+    );
+}
```
Comment (somewhatabstract) [inline utils/gen-styleguide-prod-config.js]: Same as before, but now we do it after augmenting root level sections with package info, so we can strip things we auto-marked as private.

**Comment by somewhatabstract** [comment]
@jeresig We use `prototype` in this context. If you look at where we extract the pertinent info, we grab the `private` flag from package.json and use it to set the `prototype` flag for the package as that's what it means in the context of producing the docs.

**Comment by somewhatabstract** [comment]
@jeresig see https://github.com/Khan/wonder-blocks/blob/cc090ab6e7b39ac49835880a1debae9144a47876/utils/gen-styleguide-prod-config.js#L177

**Comment by jeresig** [comment]
@somewhatabstract Ohhh - thank you for the clarification! It may be worth adding a comment to that effect, as well, just to help clarify it for others, too. Thank you!

**Pull request merged**